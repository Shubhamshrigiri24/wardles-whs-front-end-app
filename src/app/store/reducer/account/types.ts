import { TActionStatus } from "app/store/types";
import { ACCOUNT_ACTIONS } from "app/store/reducer/account/constants";

export type TAccount = {
  dob: string;
  email: string;
  firstName: string;
  gender: string;
  lastName: string;
  phone: string;
  updateTime: string;
  userId: string;
  addressline1: string;
  addressline2: string;
  agreeMarketing: boolean;
  city: string;
  gpDetails: string;
  postcode: string;
};

export type State = {
  data: TAccount | null;
  status: TActionStatus;
};

export type SetAccountDataActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT;
  payload: TAccount;
};

export type SetAccountStatusActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT_STATUS;
  payload: TActionStatus;
};

export type AccountActionTypes =
  | SetAccountDataActionType
  | SetAccountStatusActionType;
