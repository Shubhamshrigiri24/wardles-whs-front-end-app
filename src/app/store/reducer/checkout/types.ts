import { RSAA } from "redux-api-middleware";
import { PostcodeLookupAddress } from "../shared";

export interface State {
  customerDetails: CustomerDetails;
  confirmedPayment: boolean;
  braintreeToken: string;
  paymentErrorMessage: string;
  postCodeLookUpAddresses: PostcodeLookupAddress[];
  deliveryDetails: DeliveryDetails;
}

export interface CustomerDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
  email: string;
  phone: string;
  surgeryName?: string;
  surgeryAddress?: string;
  allowMarketing?: boolean;
}

export interface AddCustomerDetailsAction {
  type: string;
  payload: {
    customerDetails: CustomerDetails;
  };
}

export interface DeliveryDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
}

export interface AddDeliveryDetailsAction {
  type: string;
  payload: {
    deliveryDetails: DeliveryDetails;
  };
}

export interface FetchBraintreeTokenAction {
  type: string;
  [RSAA]: {
    endpoint: string;
    method: string;
    types: [string, string, string];
  };
}

export interface FetchBraintreeTokenResponseAction {
  type: string;
  payload: {
    token: string;
  };
}

export interface PaymentFailureAction {
  type: string;
  payload: {
    response: {
      message: string | null;
    } | null;
  };
}

export interface FetchAddressFromPostcodeActionResponse {
  type: string;
  payload: {
    addresses: PostcodeLookupAddress[];
  };
}
