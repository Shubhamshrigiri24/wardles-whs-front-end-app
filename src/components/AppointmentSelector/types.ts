export interface Availability {
  resourceId: string;
  startTime: Date;
  endTime: Date;
}

export interface FetchAvailabilityInput {
  locationId?: string;
  serviceId: string;
  start: Date;
  end: Date;
}
