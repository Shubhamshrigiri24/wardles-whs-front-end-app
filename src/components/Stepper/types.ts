export enum StepperSteps {
  CONSULTATION = "Consultation",
  PRODUCT_SELECTION = "Treatment options",
  PERSONAL_DETAILS = "Personal details",
  DELIVERY_DETAILS = "Delivery details",
  PAYMENT_DETAILS = "Payment details",
}
