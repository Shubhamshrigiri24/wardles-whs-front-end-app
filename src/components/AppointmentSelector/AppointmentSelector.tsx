import React, { useState, useEffect, useCallback } from "react";
import { Typography, Theme, makeStyles } from "@material-ui/core";
import AppointmentSlots from "./components/AppointmentSlot/AppointmentSlot";
import { Availability, FetchAvailabilityInput } from "./types";
import DatePicker from "./components/DatePicker";

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    display: "flex",
    flexWrap: "wrap",
    flexGrow: 1,
  },
  opaque: {
    opacity: 0.4,
  },
  item: {
    paddingRight: theme.spacing(3),
    paddingBottom: theme.spacing(2),
    display: "block",
  },
}));

export interface AppointmentSelectorStateProps {
  now?: Date;
  minDate?: Date;
  appointments: Availability[];
  selectedAppointment?: Availability;
  isFetching: boolean;
  selectedServiceId: string;
  selectedLocationId?: string;
}

export interface AppointmentSelectorDispatchProps {
  fetchAvailability(input: FetchAvailabilityInput): void;
  setAppointment(a: Availability): void;
}

export type AppointmentSelectorProps = AppointmentSelectorStateProps &
  AppointmentSelectorDispatchProps;

const AppointmentSelector: React.FC<AppointmentSelectorProps> = (props) => {
  const dateTimeNow = new Date();
  const initialSelectedDate: Date = props.minDate || props.now || dateTimeNow;
  const [selectedDate, setSelectedDate] = useState(initialSelectedDate);

  const {
    fetchAvailability,
    selectedLocationId,
    selectedServiceId,
    isFetching,
    appointments,
  } = props;

  const loadAvailability = useCallback(() => {
    const fetchAvailabilityInput: FetchAvailabilityInput = {
      locationId: selectedLocationId,
      serviceId: selectedServiceId,
      start: selectedDate,
      end: selectedDate,
    };
    fetchAvailability(fetchAvailabilityInput);
  }, [fetchAvailability, selectedLocationId, selectedServiceId, selectedDate]);

  useEffect(() => {
    loadAvailability();
  }, [loadAvailability, selectedDate]);

  const classes = useStyles();

  const getDaysAppointments = useCallback((): Availability[] => {
    if (!appointments?.length) {
      return [];
    }
    return props.appointments.filter(
      (x) => new Date(x.startTime).getDate() === selectedDate.getDate()
    );
  }, [props, selectedDate, appointments]);

  const onDateClick = useCallback(
    (selectedDate: Date | null) => {
      setSelectedDate(selectedDate || initialSelectedDate);
    },
    [setSelectedDate, initialSelectedDate]
  );

  const daysAppointments = getDaysAppointments();

  const getErrorMessage = useCallback(() => {
    if (!daysAppointments.length && !isFetching) {
      return "There are no available appointments on your selected date";
    }
    return "";
  }, [daysAppointments, isFetching]);

  return (
    <div className={classes.container}>
      <div className={classes.item}>
        <Typography variant={"body2"}>Select a date</Typography>
        <DatePicker
          date={selectedDate}
          minDate={props.minDate}
          errorMessageOverride={getErrorMessage()}
          handleDateChange={onDateClick}
        />
      </div>
      <div className={classes.item}>
        <Typography variant={"body2"}>Select a time</Typography>
        <AppointmentSlots
          appointments={daysAppointments}
          selectedAppointment={props.selectedAppointment}
          setAppointment={props.setAppointment}
          isLoading={isFetching}
        />
      </div>
    </div>
  );
};

export default AppointmentSelector;
