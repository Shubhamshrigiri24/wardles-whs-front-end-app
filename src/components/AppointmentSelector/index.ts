export { default as AppointmentSelector } from "./AppointmentSelector";
export * from "./types";
export * from "./AppointmentSelector";
