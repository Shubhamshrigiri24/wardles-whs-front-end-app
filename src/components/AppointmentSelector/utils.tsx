import { MaterialUiPickersDate } from "@material-ui/pickers/typings/date";
import { Availability } from "./types";

export function parseSelectedDate(date: MaterialUiPickersDate) {
  try {
    if (!date || !date?.toISOString()) {
      console.log("not valid", date);
      return null;
    }
    const parsedDate = new Date(date?.toISOString());
    return parsedDate;
  } catch {
    return null;
  }
}

export function getAppointmentKey(a: Availability) {
  const startTime = new Date(a.startTime);
  const rowKey = startTime.toISOString();
  return `${rowKey}-${a.resourceId}`;
}
