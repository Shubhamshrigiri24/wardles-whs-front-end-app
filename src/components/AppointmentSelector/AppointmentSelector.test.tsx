import React from "react";
import { render, waitForElement } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import { MemoryRouter } from "react-router-dom";
import AppointmentSelector, {
  AppointmentSelectorProps,
} from "./AppointmentSelector";
import { act } from "react-dom/test-utils";
import { Availability } from "./types";
import { getAppointmentKey } from "./utils";
const { add, format } = require("date-fns");

describe("<AppointmentSelector />", () => {
  it("should match the snapshot", async () => {
    const props = makeProps();
    const { container } = render(<AppointmentSelector {...props} />, {
      wrapper: MemoryRouter,
    });
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when loading", async () => {
    const props = makeProps();
    props.isFetching = true;
    const { container } = render(<AppointmentSelector {...props} />, {
      wrapper: MemoryRouter,
    });
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot with no availability", async () => {
    const props = makeProps();
    props.appointments = [];
    const { container } = render(<AppointmentSelector {...props} />, {
      wrapper: MemoryRouter,
    });
    expect(container).toMatchSnapshot();
  });

  it("should call fetchAvailability on load", async () => {
    const props = makeProps();
    await act(async () => {
      render(<AppointmentSelector {...props} />);
    });
    expect(props.fetchAvailability).toHaveBeenCalledTimes(1);
  });

  it("should call fetchAvailability when user enters a different date", async () => {
    const todaysDate = new Date(Date.now());
    todaysDate.setHours(0, 0, 0, 0);
    const props = makeProps({ now: todaysDate });
    const { container, getByTestId } = render(
      <AppointmentSelector {...props} />
    );
    const keyBoardDatePicker = getByTestId("keyboardDatePicker");
    expect(keyBoardDatePicker).toBeTruthy();
    const input = container.querySelector("#date-picker-dialog");
    if (!input) {
      throw new Error("could not find input");
    }
    const convertToString = add(todaysDate, { days: 1 });
    await act(async () => {
      fireEvent.change(input, {
        target: {
          value: format(new Date(add(todaysDate, { days: 1 })), "dd/MM/yyyy"),
        },
      });
    });

    expect(props.fetchAvailability).toHaveBeenCalledTimes(2);
    const calls = (props.fetchAvailability as jest.Mock<any, any>).mock.calls;
    expect(calls).toEqual([
      [
        {
          start: new Date(todaysDate),
          end: new Date(todaysDate),
          locationId: "selectedLocationId",
          serviceId: "selectedServiceId",
        },
      ],
      [
        {
          start: new Date(add(todaysDate, { days: 1 })),
          end: new Date(add(todaysDate, { days: 1 })),
          locationId: "selectedLocationId",
          serviceId: "selectedServiceId",
        },
      ],
    ]);
  });

  it("should call setAppointment when an appointment is selected", async () => {
    const props = makeProps();
    let renderObj = null;
    await act(async () => {
      const { container, getByTestId } = render(
        <AppointmentSelector {...props} />
      );
      const selectAv = props!.appointments[0];
      const key = getAppointmentKey(selectAv);
      await act(async () => {
        const select = await waitForElement(() =>
          getByTestId("appointmentSlot/appointment-select")
        );

        fireEvent.click(select, {
          target: { value: key },
        });
      });
      expect(props.setAppointment).toBeCalledWith(selectAv);
    });
  });
});

function makeProps(
  v?: Partial<AppointmentSelectorProps>
): AppointmentSelectorProps {
  return {
    now: v?.now ?? new Date("2020-12-05T12:00:00"),
    appointments: [
      makeAvailability(11),
      makeAvailability(12),
      makeAvailability(13),
    ],
    selectedAppointment: makeAvailability(12),
    isFetching: false,
    selectedServiceId: "selectedServiceId",
    selectedLocationId: "selectedLocationId",
    fetchAvailability: jest.fn(),
    setAppointment: jest.fn(),
  };
}

function makeAvailability(hour: number): Availability {
  return {
    startTime: new Date(`2020-12-05T${hour}:00:00`),
    endTime: new Date(`2020-12-05T${hour}:30:00`),
    resourceId: `123-${hour}`,
  };
}
