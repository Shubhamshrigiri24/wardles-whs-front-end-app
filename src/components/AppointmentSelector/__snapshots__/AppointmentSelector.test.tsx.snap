// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentSelector /> should match the snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="makeStyles-item-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a date
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-datePicker-4 MuiFormControl-marginNormal"
        data-testid="keyboardDatePicker"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
          for="date-picker-dialog"
          id="date-picker-dialog-label"
        >
          dd/mm/yyyy
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="date-picker-dialog"
            type="text"
            value="05/12/2020"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        
      </div>
    </div>
    <div
      class="makeStyles-item-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a time
      </p>
      <div
        class="makeStyles-root-5"
      >
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-6 MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="appointmentSlot/appointment-select"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-9"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppointmentSelector /> should match the snapshot when loading 1`] = `
<div>
  <div
    class="makeStyles-container-12"
  >
    <div
      class="makeStyles-item-14"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a date
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-datePicker-15 MuiFormControl-marginNormal"
        data-testid="keyboardDatePicker"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
          for="date-picker-dialog"
          id="date-picker-dialog-label"
        >
          dd/mm/yyyy
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="date-picker-dialog"
            type="text"
            value="05/12/2020"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        
      </div>
    </div>
    <div
      class="makeStyles-item-14"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a time
      </p>
      <div
        class="makeStyles-root-16"
      >
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17 Mui-disabled Mui-disabled MuiInputBase-formControl"
          >
            <div
              aria-disabled="true"
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled Mui-disabled"
              role="button"
            >
              <div
                class="makeStyles-loadingSpinnerCenter-18"
              >
                <div
                  class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                  role="progressbar"
                  style="width: 1em; height: 1em;"
                >
                  <svg
                    class="MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="appointmentSlot/appointment-select"
              tabindex="-1"
              value="loading"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined Mui-disabled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-19 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-20"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppointmentSelector /> should match the snapshot with no availability 1`] = `
<div>
  <div
    class="makeStyles-container-23"
  >
    <div
      class="makeStyles-item-25"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a date
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-datePicker-26 MuiFormControl-marginNormal"
        data-testid="keyboardDatePicker"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
          data-shrink="true"
          for="date-picker-dialog"
          id="date-picker-dialog-label"
        >
          dd/mm/yyyy
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-describedby="date-picker-dialog-helper-text"
            aria-invalid="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="date-picker-dialog"
            type="text"
            value="05/12/2020"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root Mui-error MuiFormHelperText-filled"
          id="date-picker-dialog-helper-text"
        >
          There are no available appointments on your selected date
        </p>
      </div>
    </div>
    <div
      class="makeStyles-item-25"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Select a time
      </p>
      <div
        class="makeStyles-root-16"
      >
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17 Mui-disabled Mui-disabled MuiInputBase-formControl"
          >
            <div
              aria-disabled="true"
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled Mui-disabled"
              role="button"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="appointmentSlot/appointment-select"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined Mui-disabled"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-27 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-28"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
